coderd/userauth_test.go

```diff
@@ -882,6 +883,92 @@ func TestUserOAuth2Github(t *testing.T) {
 		require.Equal(t, user.ID, userID, "user_id is different, a new user was likely created")
 		require.Equal(t, user.Email, newEmail)
 	})
+	t.Run("DeviceFlow", func(t *testing.T) {
```

Comment:
Just sharing some info. Our `oidctest.FakeIDP` supports device auth flow.

Here is an example external auth test:
https://github.com/coder/coder/blob/dab31057badae420c26c8dcb388601c89ca991c5/coderd/externalauth_test.go#L269-L287

So we could write a test against a "live" idp server. This test uses mocked out responses for `ExchangeDeviceCode` and `AuthorizeDevice`. It does not test any of the oauth logic behind that.